### PR TITLE
refactor(cors): support cors via rs/cors

### PR DESCRIPTION
**Pull request by YogiLiu**

### CODE REVIEW

**Comment by YogiLiu** [comment]
Hi @shellfly, it is ready for review.

File: main.go [hunk 1]
```diff
@@ -52,23 +54,28 @@ func main() {
...
 		middleware := auth.NewMiddleware([]byte(cfg.Auth.Secret))
-		http.Handle("/", middleware(restServer))
+		mux.Handle("/", middleware(restServer))
 	} else {
-		http.Handle("/", restServer)
+		mux.Handle("/", restServer)
 	}
 
+	// CORS Support
+	handler := cors.Default().Handler(mux)
```
Comment (shellfly) [inline main.go:73]: @YogiLiu do you need to make this a config item？
```
c := cors.New(cors.Options{
    AllowedOrigins: cfg.AllowedOrigins,
    AllowCredentials: true,
})
```

Comment (YogiLiu) [reply main.go]: @shellfly OK, it's done.

**Comment by shellfly** [comment]
@YogiLiu Thanks for the changes, I just made a new release of it [ v0.2.0](https://github.com/rest-go/rest/releases/tag/v0.2.0)

**Pull request merged**